backend/src/main/java/org/eclipse/tractusx/puris/backend/supply/logic/service/CustomerSupplyService.java

```diff
@@ -0,0 +1,175 @@
+
+        List<Double> ownDeliveries = ownDeliveryService.getQuantityForDays(material, partnerBpnl, siteBpns, DirectionCharacteristic.INBOUND, numberOfDays);
+        List<Double> reportedDeliveries = reportedDeliveryService.getQuantityForDays(material, partnerBpnl, siteBpns, DirectionCharacteristic.INBOUND, numberOfDays);
+        List<Double> deliveries = mergeDeliveries(ownDeliveries, reportedDeliveries);
+
+        double stockQuantity = stockService.getInitialStockQuantity(material, partnerBpnl);
+
+        for (int i = 0; i < numberOfDays; i++) {
+            Date date = Date.from(localDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+
+            if (i == numberOfDays - 1) {
+                stockQuantity += deliveries.get(i);
+            }
+
+            double daysOfSupply = getDaysOfSupply(
+                stockQuantity,
+                demands.subList(i, demands.size()));
+
+            OwnCustomerSupply supply = new OwnCustomerSupply();
+            supply.setMaterial(materialService.findByOwnMaterialNumber(material));
+            supply.setDate(date);
+            supply.setDaysOfSupply(daysOfSupply);
+            customerSupply.add(supply);
+
+            stockQuantity = stockQuantity - demands.get(i) + deliveries.get(i);
+
+            localDate = localDate.plusDays(1);
+        }
+
+        return customerSupply;
+    }
+    
+    public final List<ReportedCustomerSupply> findAll() {
+        return repository.findAll();
+    }
+
+    public final ReportedCustomerSupply findById(UUID id) {
+        return repository.findById(id).orElse(null);
+    }
+
+    public final List<ReportedCustomerSupply> findAllByFilters(Optional<String> ownMaterialNumber, Optional<String> bpnl) {
+        Stream<ReportedCustomerSupply> stream = repository.findAll().stream();
+        if (ownMaterialNumber.isPresent()) {
+            stream = stream.filter(dayOfSupply -> dayOfSupply.getMaterial().getOwnMaterialNumber().equals(ownMaterialNumber.get()));
+        }
+        if (bpnl.isPresent()) {
+            stream = stream.filter(dayOfSupply -> dayOfSupply.getPartner().getBpnl().equals(bpnl.get()));
+        }
+        return stream.toList();
+    }
+
+    public boolean validate(ReportedCustomerSupply daysOfSupply) {
+        return 
+            daysOfSupply.getPartner() != null &&
+            daysOfSupply.getMaterial() != null &&
+            daysOfSupply.getDate() != null &&
+            daysOfSupply.getStockLocationBPNS() != null &&
+            daysOfSupply.getPartner() != partnerService.getOwnPartnerEntity() &&
+            daysOfSupply.getPartner().getSites().stream().anyMatch(site -> site.getBpns().equals(daysOfSupply.getStockLocationBPNS())) &&
+            (daysOfSupply.getStockLocationBPNA() == null || daysOfSupply.getStockLocationBPNA() == daysOfSupply.getStockLocationBPNS());
+    }
+
+    private final double getDaysOfSupply(double stockQuantity, List<Double> demands) {
```

Comment:
Because first it takes the stock quantity and checks for how many days it will last depending on the demands, and then for the next iteration it adjusts the stock

Reply:
Rechecked code and standard: Beside stocks we need to consider added values (here deliveries) and consumed values (her demands)
- method currently only considers demands (actually is called with deliveries, which is wrong)
- Standard considers per day: what's added to my inventory and what's consumed
  - on first day stocks is latest stocks. 
  - on the next days: Calculate base as stock from past day + everything that will be added to my inventory and  - what has been consumed till midnight
  
We can't be 100% accurate as we have a latest stock and neither a "at midnight stock" nor an information if a delivery / production has already been considered in stock.

Reply:
aha okay.. makes sense yeah.. I'll adjust the code